### ISSUE TITLE
Parallel::Kill and Parallel::Break does not work when using `each`

### ISSUE THREAD

**Issue by mlyubarskyy** [grosser/parallel#175]
ver 1.3.1

```
#map terminates workers
Parallel.map(20.times, in_processes: 4 ) do |i|
  if i == 6
    puts "#{Process.ppid} #{$$} whooops"
    raise Parallel::Kill
  else
    sleep 1
    puts "#{Process.pid} #{$$} - #{i}"
  end
end

87600 87600 - 0
87603 87603 - 2
87601 87601 - 1
87602 87602 - 3
84925 87602 whooops
=> nil

#each continues processing
Parallel.each(20.times, in_processes: 4 ) do |i|
  if i == 6
    puts "#{Process.ppid} #{$$} whooops"
    raise Parallel::Kill
  else
    sleep 1
    puts "#{Process.pid} #{$$} - #{i}"
  end
end

87575 87575 - 2
87576 87576 - 3
87574 87574 - 0
87573 87573 - 1
84925 87576 whooops
87573 87573 - 5
87574 87574 - 4
87575 87575 - 7
87576 87576 - 8
87573 87573 - 9
87575 87575 - 10
87574 87574 - 11
87576 87576 - 12
87575 87575 - 13
87573 87573 - 15
87574 87574 - 14
87576 87576 - 16
87573 87573 - 17
87574 87574 - 18
87575 87575 - 19
```

**Comment by grosser** [comment]
dammit ... bad tests I guess ... can you track it down to an exact commit ? / add a failing testcase ?

**Comment by ndbroadbent** [comment]
This is fixed now. Out of interest, I ran `git bisect` to track down the good/bad commits. The bug was introduced in 8898c862c889d12aa20fceeb48a2d3466d768f66 and fixed in 237505d16e04e85cf65a3a6b37864b96b268d731.

**Comment by grosser** [comment]
thx for tracking this down :D
